### PR TITLE
Ruby oneofs: return default instead of nil for unset fields.

### PR DESCRIPTION
**Pull request by haberman**

### CODE REVIEW

**Comment by haberman** [comment]
Review to @acozzette.

This is part of a general decision that oneof fields should return defaults when unset instead of nil.

File: ruby/ext/google/protobuf_c/storage.c [hunk 1]
```diff
@@ -57,6 +57,28 @@ size_t native_slot_size(upb_fieldtype_t type) {
   }
 }
 
+VALUE value_from_default(const upb_fielddef *field) {
```
Comment (acozzette) [inline ruby/ext/google/protobuf_c/storage.c]: Should this function be static since it is only called from within this file?

**Comment by acozzette** [comment]
LGTM

**Comment by haberman** [comment]
I addressed your comment and also fixed JRuby. The enum behavior was slightly wrong also, so I've fixed that too. PTAL.

**Comment by acozzette** [comment]
LGTM

**Comment by seishun** [comment]
Is this the desired behavior for all languages? Because the JavaScript implementation does currently suffer from the same ailment: unset oneof fields return `undefined` (see #1447).

**Pull request merged**